--- lints/ebuild/keywords_single_line.go
+++ lints/ebuild/keywords_single_line.go
@@ -65,7 +65,7 @@
			keywordAssignCount := 0
			syntax.Walk(f, func(node syntax.Node) bool {
				if assign, ok := node.(*syntax.Assign); ok {
-					if assign.Name != nil && assign.Name.Value == "KEYWORDS" && assign.Value != nil {
+					if assign.Name != nil && assign.Name.Value == "KEYWORDS" {
						keywordAssignCount++

						// Check append
