--- lints/ebuild/nil_assignment_test.go
+++ lints/ebuild/nil_assignment_test.go
@@ -101,11 +101,31 @@
		},
		{
-			name: "KEYWORDS empty",
+			name: "KEYWORDS nil",
			content: `EAPI=8
-local KEYWORDS
-KEYWORDS=
-KEYWORDS=()
+local KEYWORDS`,
+			rule:             &KeywordsSingleLineLintRule{},
+			expectError:      false,
+			expectedFindings: 0,
+		},
+		{
+			name: "KEYWORDS empty",
+			content: `EAPI=8
+KEYWORDS=`,
+			rule:             &KeywordsSingleLineLintRule{},
+			expectError:      false,
+			expectedFindings: 0,
+		},
+		{
+			name: "KEYWORDS array",
+			content: `EAPI=8
+KEYWORDS=()`,
+			rule:             &KeywordsSingleLineLintRule{},
+			expectError:      false,
+			expectedFindings: 0,
+		},
+		{
+			name: "KEYWORDS empty quotes",
+			content: `EAPI=8
 KEYWORDS=""`,
			rule:             &KeywordsSingleLineLintRule{},
