diff --git a/cmd/g2/ebuild_upsert.go b/cmd/g2/ebuild_upsert.go
--- a/cmd/g2/ebuild_upsert.go
+++ b/cmd/g2/ebuild_upsert.go
@@ -191,3 +191,7 @@
		if newVersionOrBumpType == "revision" || newVersionOrBumpType == "" || isBumpType(newVersionOrBumpType) {
			highestRevGV.IncrementRevision()
+		} else {
+			highestRevGV = g2.ParseGentooVersion(newVersionOrBumpType)
		}
		targetFile = filepath.Join(*dirFlag, fmt.Sprintf("%s-%s.ebuild", *pkgFlag, highestRevGV.String()))
